kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/canvas/controls/BPMNCreateNodeActionTest.java

```diff
@@ -55,15 +68,50 @@ public void setUp() {
 
     @Test
     @SuppressWarnings("unchecked")
-    public void testIsAutoConnection() {
-        assertFalse(isAutoMagnetConnection(gatewayNode, taskNode));
-        assertFalse(isAutoMagnetConnection(gatewayNode, eventNode));
-        assertFalse(isAutoMagnetConnection(taskNode, gatewayNode));
-        assertFalse(isAutoMagnetConnection(eventNode, gatewayNode));
-        assertFalse(isAutoMagnetConnection(gatewayNode, gatewayNode));
-        assertTrue(isAutoMagnetConnection(taskNode, taskNode));
-        assertTrue(isAutoMagnetConnection(eventNode, taskNode));
-        assertTrue(isAutoMagnetConnection(taskNode, eventNode));
-        assertTrue(isAutoMagnetConnection(eventNode, eventNode));
+    public void testIsGatewayFixedConnection() {
+        assertTrue(isGatewaySourceOrTarget(gatewayNode, taskNode));
+        assertTrue(isGatewaySourceOrTarget(gatewayNode, eventNode));
+        assertTrue(isGatewaySourceOrTarget(taskNode, gatewayNode));
+        assertTrue(isGatewaySourceOrTarget(eventNode, gatewayNode));
+        assertTrue(isGatewaySourceOrTarget(gatewayNode, gatewayNode));
+        assertFalse(isGatewaySourceOrTarget(taskNode, taskNode));
+        assertFalse(isGatewaySourceOrTarget(eventNode, taskNode));
+        assertFalse(isGatewaySourceOrTarget(taskNode, eventNode));
+        assertFalse(isGatewaySourceOrTarget(eventNode, eventNode));
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testbuildConnectionBetween() {
```

Comment:
could we split this into multiple test cases?

Reply:
Hi @domhanak separated tests now :-)